### PR TITLE
feat: support to fetch balancer ip/port from ctx.var

### LINKED ISSUE #4091: feature: support to fetch balancer `ip`/`port` from `ctx.vars`
we have stored the upstream ip/port in `ctx` object, here is the link: https://github.com/apache/apisix/blob/master/apisix/balancer.lua#L204-L205

we should support to fetch from `ctx.vars`, here is the example:

```lua
local up_ip = ctx.vars["balancer_ip"]
local up_port = ctx.vars["balancer_port"]
```

Then we can use this value in plugins(eg: SkyWalking).

I think we can use the same way as "consumer_name", which is the code link:

https://github.com/apache/apisix/blob/master/apisix/core/ctx.lua#L176

### PR DESCRIPTION
**Pull request by Firstsawyou**
### What this PR does / why we need it:

Signed-off-by: Firstsawyou yuelinz99@gmail.com
close #4091

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: t/core/ctx2.t [hunk 1]
```diff
@@ -87,3 +87,58 @@ GET /xxx
...
+                }]]
+                )
+
+            if code >= 300 then
+                ngx.status = code
+            end
+            ngx.say(body)
+        }
+    }
+--- request
```
Comment (spacewander) [inline t/core/ctx2.t:126]: We don't need that stuff. They are already set in https://github.com/Firstsawyou/apisix/blob/a78bd10f4f1efb3547cae880d7cdea7d7ffb08e1/t/core/ctx2.t#L23

Comment (Firstsawyou) [reply t/core/ctx2.t]: Got it, updated.

File: t/core/ctx2.t [hunk 2]
```diff
@@ -87,3 +87,52 @@ GET /xxx
...
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/routes/1',
+                ngx.HTTP_PUT,
+                [[{
+                    "uri": "/hello",
+                    "plugins": {
+                        "serverless-post-function": {
+                            "phase": "log",
+                            "functions" : ["return function(conf, ctx)
+                                        ngx.log(ngx.WARN, 'balancer_ip: ', ctx.var.balancer_ip);
```
Comment (tokers) [inline t/core/ctx2.t:106]: Redundant semicolons.

**Review by spacewander — APPROVED** [review]

File: t/core/ctx2.t (hunk 2 above)
Comment (Firstsawyou) [reply t/core/ctx2.t]: Got it, updated.

**Review by tokers — APPROVED** [review]

**Pull request merged**